src/compiler/transformers/ts.ts

```diff
@@ -1793,7 +1793,7 @@ namespace ts {
                     ? getGeneratedNameForNode(name)
                     : name.expression;
             }
-            else if (isIdentifier(name)) {
+            else if (isIdentifier(name) || isPrivateName(name)) {

```

Comment:
@mheiber resolved - we return an empty identifier here (because the private identifiers can be parsed in enum members, so a user can trigger that case - the checker outputs an error)